corehq/apps/userreports/reports/filters/choice_providers.py

```diff
@@ -394,6 +397,42 @@ def default_value(self, user):
         return None
 
 
+class DomainChoiceProvider(ChainableChoiceProvider):
+
+    def _query_domains(self, domain, query_text):
```

Comment:
should this be memoized? It does get called a number in a few different places but I'm not sure how many of those are called during a request.